lib/BoundaryValueDiffEqShooting/src/multiple_shooting.jl

```diff
@@ -209,15 +233,25 @@ function __multiple_shooting_init_odecache(
 end
 
 function __multiple_shooting_init_jacobian_odecache(
-        ensemblealg, prob, jac_cache, ::NoDiffCacheNeeded, alg, nshoots, u; kwargs...)
+        ensemblealg, prob, jac_cache, diffmode, alg, nshoots, u; kwargs...)
+    __multiple_shooting_init_jacobian_odecache(
+        ensemblealg, prob, jac_cache, __cache_trait(diffmode),
+        diffmode, alg, nshoots, u; kwargs...)
+end
+
+function __multiple_shooting_init_jacobian_odecache(
+        ensemblealg, prob, jac_cache, ::NoDiffCacheNeeded,
+        diffmode, alg, nshoots, u; kwargs...)
     return __multiple_shooting_init_odecache(ensemblealg, prob, alg, u, nshoots; kwargs...)
 end
 
 function __multiple_shooting_init_jacobian_odecache(
-        ensemblealg, prob, jac_cache, ::DiffCacheNeeded, alg, nshoots, u; kwargs...)
-    cache = jac_cache.cache
-    if cache isa ForwardDiff.JacobianConfig
-        xduals = reshape(cache.duals[2][1:length(u)], size(u))
+        ensemblealg, prob, jac_cache, ::DiffCacheNeeded,
+        diffmode, alg, nshoots, u; kwargs...)
+    if diffmode isa AutoSparse
+        xduals = reshape(jac_cache.pushforward_prep.xdual_tmp[1:length(u)], size(u))
```

Comment:
This is private API of DifferentiationInterface, it can change at any time without notice, so you shouldn't touch it. The details of preparation objects are not public-facing.

Why do you need to manipulate dual numbers directly?